ledger/acctupdates.go

```diff
@@ -1911,7 +1992,16 @@ func (au *accountUpdates) lookupWithoutRewards(rnd basics.Round, addr basics.Add
 		// present in the on-disk DB.  As an optimization, we avoid creating
 		// a separate transaction here, and directly use a prepared SQL query
 		// against the database.
-		dbad, err = au.accountsq.lookup(addr)
+
+		if extension != nil {
+			dbad, err = lookupExt(au.dbs.Rdb, addr, extension)
```

Comment:
it's abit hard to follow, but I'm not sure if the extension method here is called from the *same* transaction or not. It's vital to call it from the same read transaction - otherwise, the `dbad.round` comparison below would not be sufficient.

Reply:
not sure I understand. `au.accountsq.lookup` uses prepared queries from `au.dbs.Rdb`, same way as `lookupExt` does. `lookupExt` prepares and runs two queries on the same transaction derived from `au.dbs.Rdb.Atomic`. It looks like as a complete equivalent of the old code.

Reply:
ah - I missed that. Thanks.